theano/scalar/basic.py

```diff
@@ -2299,8 +2299,8 @@ def c_code___(self, node, name, inputs, outputs, sub):
         (x,) = inputs
         (z,) = outputs
         typ = node.outputs[0].type.dtype
-        if typ not in ['float32', 'float64']:
-            Exception("The output should be float32 or float64")
+        if not typ.startswith('float'):
```

Comment:
I don't want to make scalar/basic.py depend on tensor/basic.py.  Also I guess that part can be reverted since I fixed the not-float16 problems another way.